test/distributed/test_c10d.py

```diff
@@ -3825,6 +3827,51 @@ def test_broadcast_coalesced_gloo_cpu(self):
             self._test_broadcast_coalesced(process_group, device, root_rank)
 
 
+class TestDistributedNNFunctions(MultiProcessTestCase):
+    def setUp(self):
+        super(TestDistributedNNFunctions, self).setUp()
+        os.environ['WORLD_SIZE'] = str(self.world_size)
+        os.environ['MASTER_ADDR'] = '127.0.0.1'
+        os.environ['MASTER_PORT'] = "12345"
+        # This doesnt work ...
+        os.environ['MASTER_PORT'] = str(common.find_free_port())
+        self._fork_processes()
+        os.environ['RANK'] = str(self.rank)
+
+    def tearDown(self):
+        super(TestDistributedNNFunctions, self).tearDown()
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    @property
+    def op_timeout_sec(self):
+        return 1
+
+    @property
+    def world_size(self):
+        return 2
+
+    @requires_nccl()
+    @skip_if_not_multigpu
+    def test_broadcast(self):
+        dist.init_process_group(init_method='env://', rank=self.rank, world_size=self.world_size, backend='nccl')
```

Comment:
If `find_free_port` does not work for you, you can try sth as below:

https://github.com/pytorch/pytorch/blob/f89498f3f8e9f4085428cc19bea11d3ce7852b57/test/distributed/test_c10d.py#L712-L713

Reply:
Also, right now I am facing an issue here, if I change this to use the ProcessGroup, then I get errors when I try to get the rank using `dist.get_rank()` since the WORLD process group has not been initialized.
Moreover, If I fixthis by chekcing if the group is world or not, and get the rank from within the group,
the processes hang when I do group.broadcast ...
Anby clue about this? I have been stuck in here for a while ...

Reply:
I don't know why but it started to work 😂

Reply:
> if I change this to use the ProcessGroup, then I get errors when I try to get the rank using dist.get_rank()

You can call `pg.rank()` to get the rank from the `ProcessGroup` instance if that's option.

https://github.com/pytorch/pytorch/blob/9c8078cdfbe32d38d9ea5568a47174aee8e3a5f7/torch/csrc/distributed/c10d/init.cpp#L635